test/foxStakingTest.ts

```diff
@@ -1,20 +1,25 @@
-import { ethers, deployments, getNamedAccounts } from "hardhat";
+import { ethers, deployments, getNamedAccounts, network } from "hardhat";
 import { expect } from "chai";
 import { Foxy } from "../typechain-types/Foxy";
 import { FoxStaking } from "../typechain-types/FoxStaking";
 import { StakingWarmup } from "../typechain-types/StakingWarmup";
-import { ERC20 } from "../typechain-types/ERC20";
 import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
-import { BigNumber, Signer } from "ethers";
+import { BigNumber, Contract, Signer } from "ethers";
+import { foxAbi } from "./fox-abi";
+import { ERC20 } from "../typechain-types";
 
 describe("FoxStaking", function () {
   let accounts: SignerWithAddress[];
   let FOXy: Foxy;
   let foxStaking: FoxStaking;
-  let fox: ERC20;
+  let fox: Contract;
   let stakingWarmup: StakingWarmup;
 
+  const FOX_WHALE = "0xF152a54068c8eDDF5D537770985cA8c06ad78aBB"; // Keep updated with a whale's FOX address.  Address must have ETH
```

Comment:
This seems a bit fickle to rely on an on-chain address not changing for our tests to work.

Given that FOX is a generic ERC20, I think it would be better to just deploy a fake ERC20 that we entirely control for the durability of the tests. 

This also helps because we want this to work with any ERC20 in the future.

Reply:
Solved by setting block number